homeassistant/components/daikin/number.py

```diff
@@ -0,0 +1,146 @@
+"""Support for Daikin AirBase zone temperatures."""
+from __future__ import annotations
+
+from collections.abc import Callable
+
+from homeassistant.components.number import NumberDeviceClass, NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, Platform, UnitOfTemperature
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_state_change_event
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from . import DOMAIN as DAIKIN_DOMAIN, DaikinApi
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up Daikin climate based on config_entry."""
+    daikin_api: DaikinApi = hass.data[DAIKIN_DOMAIN][entry.entry_id]
+    if zones := daikin_api.device.zones:
+        async_add_entities(
+            [
+                DaikinZoneTemperature(daikin_api, zone_id)
+                for zone_id, zone in enumerate(zones)
+                if zone[0] != "-" and zone[2] != 0
+            ]
+        )
+
+
+class DaikinZoneTemperature(NumberEntity):
+    """Representation of a zone temperature setting."""
+
+    _attr_icon = "mdi:thermostat"
+    _attr_native_step = 1
+    _attr_has_entity_name = True
+    _attr_device_class = NumberDeviceClass.TEMPERATURE
+    _attr_native_unit_of_measurement = UnitOfTemperature.CELSIUS
+    _registry: er.EntityRegistry | None
+    _remove_listener: Callable[[], None] | None
+    _target_temperature: float
+    _previous_value: float | None
+    _current_value: float
+
+    def __init__(self, daikin_api: DaikinApi, zone_id) -> None:
+        """Initialize the zone."""
+        self._api = daikin_api
+        self._zone_id = zone_id
+        self._registry = None
+        self._remove_listener = None
+        self._target_temperature = self._api.device.target_temperature
+        self._previous_value = None
+        self._current_value = self._api.device.zones[self._zone_id][2]
+        self._attr_device_info = self._api.device_info
+        self._attr_unique_id = f"{self._api.device.mac}-zone-temp{self._zone_id}"
+        self._attr_name = f"{self._api.device.zones[self._zone_id][0]} temperature"
+        self._attr_native_min_value = self._target_temperature - 2
+        self._attr_native_max_value = self._target_temperature + 2
+
+    @property
+    def native_value(self) -> float:
```

Comment:
This whole thing with a previous value and a new value etc. makes no sense.
Can't you copy the logic from `sensor.py` and make some thing similar.